file_scanning/clamd/conn.go

```diff
@@ -0,0 +1,143 @@
+package clamd
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"net"
+	"regexp"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+)
+
+const CHUNK_SIZE = 1024
+const TCP_TIMEOUT = time.Second * 2
+
+var resultRegex = regexp.MustCompile(
+	`^(?P<path>[^:]+): ((?P<desc>[^:]+)(\((?P<virhash>([^:]+)):(?P<virsize>\d+)\))? )?(?P<status>FOUND|ERROR|OK)$`,
+)
+
+type CLAMDConn struct {
```

Comment:
All exported types/functions in Go are what would be classed as `public` in a language like PHP. Therefore Go says they should all have comments to give an explanation on what it does because technically it's a public facing API and therefore can be used by other packages.

`(can be ignored)`